pyronear/datasets/openfire.py

```diff
@@ -42,13 +54,16 @@ class OpenFire(VisionDataset):
     seed = 42
 
     def __init__(self, root, train=True, transform=None, target_transform=None,
-                 download=False, threads=16, valid_pct=None):
+                 download=False, threads=16, valid_pct=0.2, num_samples=None):
         super(OpenFire, self).__init__(root, transform=transform,
                                     target_transform=target_transform)
         self.train = train  # training set or test set
 
-        if download:
-            self.download(threads, valid_pct)
+        if not (self._check_exists(train=True) and self._check_exists(train=False)):
+            self.create_directories(num_samples=num_samples)
+            if download:
+                self.download(threads)
+            self.check_images(valid_pct=valid_pct)
```

Comment:
We have to think a bit much about this:
if you provide a different value of  `valid_pct` with `download=True`, currently, the condition of  `self._check_exists` will not be verified. So it will use extract that was created with a different value of `valid_pct`